src/features/dashboard/components/StatCards.tsx

```diff
@@ -1,102 +1,160 @@
 import { formatNumber } from '@/utils/formatNumber';
 import { CardData, StatCard } from './StatCard';
-import { Loader as LoaderIcon } from 'lucide-react';
+import {
+  Wallet,
+  TrendingUp,
+  TrendingDown,
+  PiggyBank,
+  Coins,
+  Receipt,
+  BarChart3,
+} from 'lucide-react';
 import { useStatCardData } from '@/hooks/useStatData';
+import { LoadingState } from '@/components/LoadingState';
 
 export const StatCards = () => {
   const { isLoading, statData, calculatedValues } = useStatCardData();
 
+  const getProfitColorAndIcon = (value: number) => {
+    const isPositive = value >= 0;
+    return {
+      icon: isPositive ? (
+        <TrendingUp className="h-5 w-5 text-[hsl(var(--chart-2))]" />
+      ) : (
+        <TrendingDown className="h-5 w-5 text-destructive" />
+      ),
+      valueColor: isPositive
+        ? 'text-[hsl(var(--chart-2))]'
+        : 'text-destructive',
+    };
+  };
+
+  const unrealizedProfitStyle = getProfitColorAndIcon(
+    statData.unrealizedProfit,
+  );
+  const realizedProfitStyle = getProfitColorAndIcon(statData.realizedProfit);
```

Comment:
Icon and color style should be separately calculated. If you need to use "and" in any given naming it's a good indicator that it should be split. In this case it could be:
```tsx
  const getProfitIcon = (value: number) =>
  value >= 0 ? (
    <TrendingUp className="h-5 w-5 text-[hsl(var(--chart-2))]" />
  ) : (
    <TrendingDown className="h-5 w-5 text-destructive" />
  );

  const profitStyle = (value: number) =>
  value >= 0 ? 'text-[hsl(var(--chart-2))]' : 'text-destructive';

  const UnrealizedProfitIcon = getProfitIcon(statData.unrealizedProfit);
  const RealizedProfitIcon = getProfitIcon(statData.realizedProfit);

  const unrealizedColorStyle = profitStyle(statData.unrealizedProfit);
  const realizedProfitStyle = profitStyle(statData.realizedProfit);
```